src/main/java/shon/task/Deadline.java

```diff
@@ -0,0 +1,62 @@
+package shon.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline task in the <code>TaskList</code>.
+ */
+public class Deadline extends Task {
+    /** The by datetime of the task */
+    private LocalDateTime by;
+    /** The formatter for reading/storing the by datetime */
+    private static DateTimeFormatter ioFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```

Comment:
Curious if you really need static

Reply:
I included static for the formatter so that all deadlines will only use the one common formatter so that the date format of all Deadline objects are the same. If not set as static, each deadline could technically have it's own formatter, and the output might be different for different Deadline objects, which is not ideal.